options.go

```diff
@@ -301,3 +305,41 @@ func parseVsocks(devices []string) ([]firecracker.VsockDevice, error) {
 func createFifoFileLogs(fifoPath string) (*os.File, error) {
 	return os.OpenFile(fifoPath, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
 }
+
+// Looks at directorys in order:
+//	* $HOME
+//	* os.TempDir()
+//
+// If none of these exist, panics
+func getSocketPath() string {
+	filename := strings.Join([]string{
+		".firecracker.sock",
+		strconv.Itoa(os.Getpid()),
+		strconv.Itoa(rand.Intn(1000))},
+		"-",
+	)
+	var dir string
+	if d := os.Getenv("HOME"); checkExistsAndDir(d) {
+		dir = d
+	} else if checkExistsAndDir(os.TempDir()) {
+		dir = os.TempDir()
+	} else {
+		panic("It's not going to do any good to land on Mars if we're stupid")
```

Comment:
:+1: , though I worry it might be misconstrued. Should probably at least mention that the failure is due to being unable to figure out where to place the socket.

Reply:
Maybe credit Ray Bradbury so it's not misinterpreted? :laughing:

Reply:
Did both of these!